jf_agent/git/gitlab_adapter.py

```diff
@@ -232,79 +236,79 @@ def get_pull_requests(
         print('downloading gitlab prs... ', end='', flush=True)
 
         for i, nrm_repo in enumerate(normalized_repos, start=1):
-            print(f'downloading prs for repo {nrm_repo.name} ({nrm_repo.id})')
+            print(f'\ndownloading prs for repo {nrm_repo.name} ({nrm_repo.id})')
 
             with agent_logging.log_loop_iters(logger, 'repo for pull requests', i, 1):
                 try:
                     pull_since = pull_since_date_for_repo(
                         server_git_instance_info, nrm_repo.project.login, nrm_repo.id, 'prs'
                     )
+                    api_prs = self.git_client.list_project_merge_requests(nrm_repo.id)
 
-                    api_prs = self.client.list_project_merge_requests(nrm_repo.id)
-
+                    api_total = api_prs.total if self.git_client.version == '4' else len(api_prs)
                     for api_pr in tqdm(
                         api_prs,
                         desc=f'processing prs for {nrm_repo.name} ({nrm_repo.id})',
                         unit='prs',
-                        total=api_prs.total,
+                        total=api_total,
                     ):
                         try:
                             updated_at = parser.parse(api_pr.updated_at)
-
                             # PRs are ordered newest to oldest
                             # if this is too old, we're done with this repo
                             if pull_since and updated_at < pull_since:
                                 break
 
+                            # verify source project is accessible
                             try:
-                                api_pr = self.client.expand_merge_request_data(api_pr)
+                                api_pr = self.git_client.expand_merge_request_data(api_pr)
                             except MissingSourceProjectException as e:
                                 log_and_print_request_error(
                                     e,
                                     f'fetching source project {api_pr.source_project_id} '
                                     f'for merge_request {api_pr.id}. Skipping...',
                                 )
                                 continue
-
                             nrm_commits: List[NormalizedCommit] = [
                                 _normalize_commit(
                                     commit,
                                     nrm_repo,
                                     self.config.git_strip_text_content,
                                     self.config.git_redact_names_and_urls,
+                                    self.git_client.version,
                                 )
                                 for commit in api_pr.commit_list
                             ]
-                            merge_request = self.client.expand_merge_request_data(api_pr)
-                            merge_commit = None
+                            merge_commit = []
```

Comment:
Doesn't it make more sense to keep the default `merge_commit = None`? (Because this is holding a single thing not a list of things.)

Reply:
It was creating downstream TypeErrors which was why I changed it, the empty list has the same significance to the conditional check of `if merge_commit` in the adapter.